metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/ConnectorFactoryDecorator.java

```diff
@@ -74,12 +93,18 @@ public ConnectorDatabaseService getDatabaseService() {
     public ConnectorTableService getTableService() {
         ConnectorTableService service = delegate.getTableService();
 
-        if (rateLimiterEnabled) {
-            log.info("Creating rate-limited connector table services for connector-type: {}, "
-                         + "plugin-type: {}, catalog: {}, shard: {}",
+        if (validationEnabled) {
+            log.info("Creating validating connector table services for connector-type: {}, "
+                    + "plugin-type: {}, catalog: {}, shard: {}",
                 connectorContext.getConnectorType(), connectorPlugin.getType(),
                 connectorContext.getCatalogName(), connectorContext.getCatalogShardName());
-            service = new ThrottlingConnectorTableService(service, rateLimiter);
+            service = new ValidatingConnectorTableService(
+                service,
+                rateLimiter,
+                rateLimiterEnabled,
+                authorization,
+                authorizationEnabled
```

Comment:
The authorization object should have the capability to determine its own enabled status, rather than having it flagged externally.